[Reverted] [WFLY-6779] added WISE feature code

rsearls opened the pull request:
Kabir: This PR was merged and then reverted as mentioned in https://issues.jboss.org/browse/WFLY-6779.

Commit:
[WFLY-6779] added WISE feature code

kabir commented on feature-pack/src/main/resources/configuration/standalone/subsystems.xml:
```diff
@@ -30,5 +30,6 @@
         <subsystem>undertow.xml</subsystem>
         <subsystem>webservices.xml</subsystem>
         <subsystem>weld.xml</subsystem>
+	<subsystem>wise.xml</subsystem>
```
Nitpick: indentation

kabir commented on pom.xml:
```diff
@@ -307,6 +309,7 @@
         <module>servlet-feature-pack</module>
         <module>webservices</module>
         <module>weld</module>
+	    <module>wise</module>
```
Nitpick: indentation

kabir commented:
I've not looked in much detail, but notice that this needs an xsd. jdr/jboss-as-jdr/src/main/resources/schema/jboss-as-jdr_1_0.xsd contains an example for an empty subsystem.

ctomc commented on feature-pack/src/main/resources/content/appclient/configuration/appclient.xml:
```diff
@@ -154,6 +154,13 @@
+                            <module-option name="realm" value="ManagementRealm"/>
+                        </login-module>
+                    </authentication>
+                </security-domain>
```
app client needs this? really?

rsearls replied:
Just to be clear, I should move this into ./security/subsystem/src/main/resources/subsystem-templates/security.xml
Right?

----- Original Message -----
> From: "Tomaz Cerar" <notifications@github.com>
> To: "wildfly/wildfly" <wildfly@noreply.github.com>
> Cc: "rsearls" <rsearls@redhat.com>, "Author" <author@noreply.github.com>
> Sent: Tuesday, July 5, 2016 5:53:11 AM
> Subject: Re: [wildfly/wildfly] [WFLY-6779] added WISE feature code (#8998)
> 
> > @@ -154,6 +154,13 @@
> >                          <policy-module code="Client" flag="required"/>
> >                      </authorization>
> >                  </security-domain>
> > +                <security-domain name="wise-security-domain"
> > cache-type="default">
> > +                    <authentication>
> > +                        <login-module code="RealmDirect" flag="required">
> > +                            <module-option name="realm"
> > value="ManagementRealm"/>
> > +                        </login-module>
> > +                    </authentication>
> > +                </security-domain>
> 
> app client needs this? really?
> 
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/wildfly/wildfly/pull/8998/files/1cc1e6d45243b24d555028c72475d8565505c969#r69535902

ctomc replied:
yes

ctomc commented on pom.xml:
```diff
@@ -6074,6 +6077,26 @@
+                    </exclusion>
+                    <exclusion>
+                        <groupId>org.jboss.as</groupId>
+                        <artifactId>jboss-as-controller</artifactId>
```
no need to exclude that, as it shouldn't be used in first place at all.

ctomc commented on servlet-feature-pack/pom.xml:
```diff
@@ -376,7 +376,18 @@
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
```
why servlet feature pack? does this work without web services present?

ctomc commented on servlet-feature-pack/pom.xml:
```diff
@@ -428,6 +439,36 @@
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <version>2.10</version>
```
this needs to be completely removed, as feature pack wont work like this.
it is maven dependency and such should be defined in module.xml provisioning plugin does the rest of the work

ctomc commented on servlet-feature-pack/src/main/resources/modules/system/layers/base/org/jboss/as/wise/main/module.xml:
```diff
@@ -0,0 +1,42 @@
+  ~ 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+  -->
+
+<module xmlns="urn:jboss:module:1.1" name="org.jboss.as.wise">
```
1.3
name, org.wildfly.extension.wise, we don't use org.jboss.as namespace for new stuff anymore.

ctomc commented on wise/pom.xml:
```diff
@@ -0,0 +1,97 @@
+        <version>10.1.0.Final-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>jboss-as-wise</artifactId>
```
this should be wildfly-wise

ctomc commented on wise/pom.xml:
```diff
@@ -0,0 +1,97 @@
+        <dependency>
+            <groupId>org.jboss.as</groupId>
+            <artifactId>jboss-as-controller</artifactId>
+            <version>7.2.0.Final</version>
```
remove this dependancy completly, as it brings in different version that is already build.

just use 
```
<dependency>
           <groupId>org.wildfly.core</groupId>
           <artifactId>wildfly-controller</artifactId>
</dependency>
```

doing this will also remove need for extra exclusions everywhere.

ctomc commented on wise/pom.xml:
```diff
@@ -0,0 +1,97 @@
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
```
remove it completely , as config like this is no longer needed.

ctomc commented on wise/src/main/java/org/jboss/as/wise/WiseExtension.java:
```diff
@@ -0,0 +1,144 @@
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.as.wise;
```
package name should be
org.wildfly.extension.wise

ctomc commented on wise/src/main/java/org/jboss/as/wise/WiseExtension.java:
```diff
@@ -0,0 +1,144 @@
+   private static final int MANAGEMENT_API_MAJOR_VERSION = 1;
+   private static final int MANAGEMENT_API_MINOR_VERSION = 0;
+   private static final int MANAGEMENT_API_MICRO_VERSION = 0;
+
```
use private static final ModelVersion VERSION = ModelVersion.create(1,0,0) instead.

ctomc commented on wise/src/main/java/org/jboss/as/wise/WiseExtension.java:
```diff
@@ -0,0 +1,144 @@
+   @Override
+   public void initialize(final ExtensionContext context) {
+      WiseLogger.ROOT_LOGGER.activatingWiseExtension();
+      final SubsystemRegistration subsystem = context.registerSubsystem(
```
should be 
context.registerSubsystem(VERSION);

ctomc commented on wise/src/main/java/org/jboss/as/wise/WiseLogger.java:
```diff
@@ -0,0 +1,46 @@
+   WiseLogger ROOT_LOGGER = Logger.getMessageLogger(WiseLogger.class, "org.jboss.as.wise");
+
+   @LogMessage(level = INFO)
+   @Message(id = 1, value = "Activating Wise Extension")
```
this should be level DEBUG, not INFO

ctomc commented on wise/src/main/java/org/jboss/as/wise/WiseSubsystemAdd.java:
```diff
@@ -0,0 +1,98 @@
+         Module module = Module.forClass(getClass());
+         URL url = module.getExportedResource("wise.war");
+         String urlString = url.toExternalForm();
+         System.out.println(url);
```
system.out must not be used.

ctomc commented:
on top of all stuff mentioned in code review, this PR is missing at least subsystem test case.

rsvoboda commented on feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/extension/wise/main/module.xml:
```diff
@@ -0,0 +1,42 @@
+
+<!--
+  ~ JBoss, Home of Professional Open Source.
+  ~ Copyright 2010, Red Hat, Inc., and individual contributors
```
year ...

rsvoboda commented on wise/pom.xml:
```diff
@@ -0,0 +1,83 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ JBoss, Home of Professional Open Source.
+  ~ Copyright 2011, Red Hat, Inc., and individual contributors
```
year ...

rsvoboda commented on wise/src/main/java/org/wildfly/extension/wise/WiseExtension.java:
```diff
@@ -0,0 +1,142 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2013, Red Hat, Inc., and individual contributors
```
year ...

rsvoboda commented on wise/src/main/java/org/wildfly/extension/wise/WiseExtension.java:
```diff
@@ -0,0 +1,142 @@
+/**
+ * Extension for subsystem that deploys wise.war.
+ *
+ * @author Brian Stansberry (c) 2012 Red Hat Inc.
```
author

rsvoboda commented on wise/src/main/java/org/wildfly/extension/wise/WiseLogger.java:
```diff
@@ -0,0 +1,46 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2013, Red Hat, Inc., and individual contributors
```
year ...

rsvoboda commented on wise/src/main/java/org/wildfly/extension/wise/WiseMessages.java:
```diff
@@ -0,0 +1,36 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2013, Red Hat, Inc., and individual contributors
```
year ...

rsvoboda commented on wise/src/main/java/org/wildfly/extension/wise/WiseSubsystemAdd.java:
```diff
@@ -0,0 +1,97 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2013, Red Hat, Inc., and individual contributors
```
year ...

rsvoboda commented on wise/src/main/java/org/wildfly/extension/wise/WiseSubsystemRemove.java:
```diff
@@ -0,0 +1,68 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2013, Red Hat, Inc., and individual contributors
```
year ...

rsvoboda commented on wise/src/main/resources/schema/jboss-as-wise_1_0.xsd:
```diff
@@ -0,0 +1,33 @@
+
+<!--
+  ~ JBoss, Home of Professional Open Source.
+  ~ Copyright 2011, Red Hat, Inc., and individual contributors
```
year ...

rsvoboda commented on wise/src/test/java/org/wildfly/extension/wise/WiseSubsystemTestCase.java:
```diff
@@ -0,0 +1,53 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2013, Red Hat, Inc., and individual contributors
```
year ...

rsvoboda commented on wise/src/test/java/org/wildfly/extension/wise/WiseSubsystemTestCase.java:
```diff
@@ -0,0 +1,53 @@
+import org.jboss.as.subsystem.test.AbstractSubsystemBaseTest;
+
+/**
+ * @author rsearls@redhat.com
```
Othe places use  * User: rsearls
please sync it

rsvoboda commented on wise/src/test/java/org/wildfly/extension/wise/WiseSubsystemTestCase.java:
```diff
@@ -0,0 +1,53 @@
+
+    @Override
+    protected String getSubsystemXsdPath() throws Exception {
+        return "schema/jboss-as-wise_1_0.xsd";
```
Shouldn't be the schema named wildfly-wise-... ?

ctomc commented on wise/src/main/java/org/wildfly/extension/wise/WiseMessages.java:
```diff
@@ -0,0 +1,36 @@
+
+/**
+ */
+@MessageBundle(projectCode = "JBWS")
```
probably WFLYWSW

ctomc commented on feature-pack/pom.xml:
```diff
@@ -89,6 +89,33 @@
+                                    <type>war</type>
+                                    <overWrite>true</overWrite>
+                                    <outputDirectory>${project.build.directory}/wildfly-${project.version}/modules/system/layers/base/org/wildfly/extension/wise/main/deployments</outputDirectory>
+                                    <destFileName>wise.war</destFileName>
```
i think this can be done in feature pack config file (feature-pack-build.xml)
with <copy-artifacts> notion which would than copy the war file where ever you want it to.

ctomc commented:
Beyond few small things commented above, it looks ok.
Just please squash commits into just one once you are done.

rsvoboda commented:
Can be this merged ?

rsearls commented:
All the issues identified since original submission have been addressed.   
This should be mergable

ctomc commented on wise/src/main/java/org/wildfly/extension/wise/WiseLogger.java:
```diff
@@ -0,0 +1,46 @@
+
+   @LogMessage(level = DEBUG)
+   @Message(id = 1, value = "Activating Wise Extension")
+   void activatingWiseExtension();
```
given that this is logged at DEBUG level, which is proper thing, it shouldn't be localized and as such no need to have it in localized logger.

ctomc commented on wise/src/main/java/org/wildfly/extension/wise/WiseMessages.java:
```diff
@@ -0,0 +1,36 @@
+   /**
+    * The messages
+    */
+   WiseMessages MESSAGES = Messages.getBundle(WiseMessages.class);
```
please remove this one, as having one localized logger is enough.
Just keep WiseLogger as that one should do it.
We also changed all subsystems to have all localized stuff in one place instead in two.

ctomc commented on wise/src/main/java/org/wildfly/extension/wise/WiseSubsystemAdd.java:
```diff
@@ -0,0 +1,97 @@
+   }
+
+   @Override
+   protected void populateModel(ModelNode operation, ModelNode model)
```
you don't need to override this method, just remove it all together.

ctomc commented:
I left few minor comments which are mostly just cleanup not major changes in the code.
The only thing that I am not 100% about is, does this subsystem really need to be in default config (standalone.xml) as I think it would be better suited in standalone-full.xml
But none of this is showstopper.

rsearls commented:
It should be in both standalone.xml and standalone-full.xml.  Is there some other file to be edited to add to 
standalone-full?

ctomc commented:
@rsearls subsystems-full.xml or subsystems-*.xml

rsearls commented:
removed WiseMessages.java
removed 2nd WiseSubsystemAdd.populateModel() method
removed WiseLogger.activatingWiseExtension()
Added wise to standalone-full.xml

rsearls commented:
I don't see that any of the wise code or configuration cause this failure.
Any suggestions on how best to resolve?

kabir commented:
Retest this please
